alacritty/src/event.rs

```diff
@@ -635,38 +635,46 @@ impl<'a, N: Notify + 'a, T: EventListener> input::ActionContext<T> for ActionCon
             return;
         }
 
-        match &hint.action {
+        let (target, bounds) = match hint {
+            HintMatch::Hyperlink { hyperlink, .. } => (hyperlink.uri().to_owned(), None),
+            HintMatch::Regex { bounds, .. } => {
+                (self.terminal.bounds_to_string(*bounds.start(), *bounds.end()), Some(bounds))
+            },
+        };
+
+        match &hint.action() {
             // Launch an external program.
             HintAction::Command(command) => {
-                let text = self.terminal.bounds_to_string(*hint.bounds.start(), *hint.bounds.end());
                 let mut args = command.args().to_vec();
-                args.push(text);
+                args.push(target);
                 start_daemon(command.program(), &args);
             },
             // Copy the text to the clipboard.
             HintAction::Action(HintInternalAction::Copy) => {
-                let text = self.terminal.bounds_to_string(*hint.bounds.start(), *hint.bounds.end());
-                self.clipboard.store(ClipboardType::Clipboard, text);
+                self.clipboard.store(ClipboardType::Clipboard, target);
             },
             // Write the text to the PTY/search.
             HintAction::Action(HintInternalAction::Paste) => {
-                let text = self.terminal.bounds_to_string(*hint.bounds.start(), *hint.bounds.end());
-                self.paste(&text);
+                self.paste(&target);
             },
             // Select the text.
             HintAction::Action(HintInternalAction::Select) => {
-                self.start_selection(SelectionType::Simple, *hint.bounds.start(), Side::Left);
-                self.update_selection(*hint.bounds.end(), Side::Right);
-                self.copy_selection(ClipboardType::Selection);
+                if let Some(bounds) = bounds {
+                    self.start_selection(SelectionType::Simple, *bounds.start(), Side::Left);
+                    self.update_selection(*bounds.end(), Side::Right);
+                    self.copy_selection(ClipboardType::Selection);
+                }
```

Comment:
I think it's unfortunate that hyperlinks can't be selected. I'd assume the primary reason for this is that a hyperlink might not be continuous?

Reply:
Yes. But we could save their location when scanning hyperlinks in `HintState::update`, which makes the first part selectable. It should be fine for most cases, yes?

Reply:
Not sure if we should offer it if it doesn't reliably make sense. Maybe selecting from the start of the first part to the end of the last part makes sense? But that could involve stuff like tmux panes and cause strange issues.

Reply:
It's now supported.